### PR TITLE
Book: fix typos/grammar, markdown lint

### PR DESCRIPTION
**Pull request by mattkdab**
Read through the book to look for needed edits and verify it was easy to follow / made sense.
Made small edits: typos, grammar, simplifying wording
Fixed sensible things that markdownlint complained about for consistency, readability, and style conventions

### CODE REVIEW

File: book/src/SUMMARY.md [hunk 1]
```diff
@@ -9,20 +9,20 @@ SPDX-License-Identifier: MIT OR Apache-2.0
...
-    - [Creating the QML GUI](./getting-started/3-qml-gui.md)
-    - [Building with Cargo](./getting-started/4-cargo-executable.md)
-    - [Building with CMake](./getting-started/5-cmake-integration.md)
+  - [`QObject`s in Rust](./getting-started/1-qobjects-in-rust.md)
+  - [Our first CXX-Qt module](./getting-started/2-our-first-cxx-qt-module.md)
+  - [Creating the QML GUI](./getting-started/3-qml-gui.md)
+  - [Building with Cargo](./getting-started/4-cargo-executable.md)
+  - [Building with CMake](./getting-started/5-cmake-integration.md)
 - [Core Concepts](./concepts/index.md)
-    - [Build Systems](./concepts/build_systems.md)
```
Comment (ahayzen-kdab) [inline book/src/SUMMARY.md:18]: what is markdown lint and can we run it in CI ?

Comment (mattkdab) [reply book/src/SUMMARY.md:18]: from the [VSCode plugin page](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) 
>Linting is performed by the [markdownlint-cli2 engine](https://github.com/DavidAnson/markdownlint-cli2), which can be used in conjunction with this extension to provide command-line support for scripts and continuous integration scenarios. The [markdownlint-cli2-action GitHub Action](https://github.com/marketplace/actions/markdownlint-cli2-action) uses the same engine and can be integrated with project workflows.

Comment (LeonMatthesKDAB) [reply book/src/SUMMARY.md:18]: I've opened an issue for this. We can always add that to CI later :)

**Review by LeonMatthesKDAB — APPROVED** [review]
It's a bit too much to check every minute detail, but overall looks like an improvement :+1:

**Pull request merged**